src/web/pages/credentials/CredentialComponent.tsx

```diff
@@ -0,0 +1,243 @@
+/* SPDX-FileCopyrightText: 2024 Greenbone AG
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import {useState} from 'react';
+import {useSelector} from 'react-redux';
+import {type CredentialDownloadFormat} from 'gmp/commands/credential';
+import {type EntityCommandParams} from 'gmp/commands/entity';
+import {
+  ALL_CREDENTIAL_TYPES,
+  type SNMPAuthAlgorithmType,
+  type default as Credential,
+  type CredentialType,
+  type SNMPPrivacyAlgorithmType,
+} from 'gmp/models/credential';
+import {isDefined} from 'gmp/utils/identity';
+import {shorten} from 'gmp/utils/string';
+import useEntityClone, {
+  type EntityCloneResponse,
+} from 'web/entity/hooks/useEntityClone';
+import useEntityCreate, {
+  type EntityCreateResponse,
+} from 'web/entity/hooks/useEntityCreate';
+import useEntityDelete from 'web/entity/hooks/useEntityDelete';
+import useEntityDownload, {
+  type OnDownloadedFunc,
+} from 'web/entity/hooks/useEntityDownload';
+import useEntitySave, {
+  type EntitySaveResponse,
+} from 'web/entity/hooks/useEntitySave';
+import useGmp from 'web/hooks/useGmp';
+import useShallowEqualSelector from 'web/hooks/useShallowEqualSelector';
+import useTranslation from 'web/hooks/useTranslation';
+import CredentialDialog, {
+  type CredentialDialogState,
+} from 'web/pages/credentials/CredentialDialog';
+import {getUserSettingsDefaults} from 'web/store/usersettings/defaults/selectors';
+import {getUsername} from 'web/store/usersettings/selectors';
+import {generateFilename} from 'web/utils/Render';
+
+interface CredentialComponentRenderProps {
+  clone: (credential: Credential) => void;
+  create: () => void;
+  delete: (credential: Credential) => void;
+  edit: (credential: Credential) => void;
+  download: (credential: Credential) => void;
+  downloadInstaller: (
+    credential: Credential,
+    format: CredentialDownloadFormat,
+  ) => Promise<void>;
+}
```

Comment:
I think there is a mismatch.
```suggestion
interface CredentialComponentRenderProps {
  clone: (credential: Credential) => Promise<void>;
  create: () => void;
  delete: (credential: Credential) => Promise<void>;
  edit: (credential: Credential) => void;
  download: (credential: Credential) => Promise<void>;
  downloadInstaller: (
    credential: Credential,
    format: CredentialDownloadFormat,
  ) => Promise<void>;
}
```